src/matlab2tikz.m

```diff
@@ -3005,6 +3008,61 @@ case size(cData,2) % specific y-data is given
     end
 end
 % ==============================================================================
+function [m2t, str] = drawHistogram(m2t, h)
+    
+    if ~isVisible(h)
+        str = '';
+        return;
+    end
+
+    % Init options 
+    opts = opts_new();
+    
+    % Face
+    faceColor = get(h,'FaceColor');
+    if isNone(faceColor)
+        opts = opts_add(opts, 'fill', 'none');
+    else
+        [m2t, xFaceColor] = getColor(m2t, h, faceColor, 'patch');
+        opts = opts_add(opts, 'fill', xFaceColor);
+    end
+    
+    % FaceAlpha
+    faceAlpha = get(h, 'FaceAlpha');
+    if ~isNone(faceColor) && isnumeric(faceAlpha) && faceAlpha ~= 1.0
+        opts = opts_add(opts,'fill opacity', sprintf(m2t.ff,faceAlpha));
+    end
+    
+    % Edge
+    edgeColor = get(h, 'EdgeColor');
+    if isNone(edgeColor)
+        opts = opts_add(opts, 'draw', 'none');
+    else
+        [m2t, xEdgeColor] = getColor(m2t, h, edgeColor, 'patch');
+        opts = opts_add(opts, 'draw', xEdgeColor);
+    end
+    
+    % Bar type
+    opts = opts_add(opts, 'ybar interval');
+    
+    % Data
+    binEdges = get(h, 'BinEdges');
+    binValue = get(h, 'Values');
+    data     = [binEdges(:), [binValue(:); binValue(end)]];
+        
+    % Make table
+    [m2t, table] = makeTable(m2t, {'x','y'},data);
+    
+    % Add 'area legend' to the options as otherwise the legend indicators
+    % will just highlight the edges.
+    m2t.axesContainers{end}.options = ...
+        opts_add(m2t.axesContainers{end}.options, 'area legend');
```

Comment:
Can't these be added to the actual entries instead of the axis? Because adding it to the axis will certainly cause other plots to become "area legends" as well.